src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -498,11 +532,21 @@ private CodeType CreateModelClasses(OpenApiUrlSpaceNode rootNode, OpenApiUrlSpac
             if(existingClass == null) // we can find it in the components
             {
                 existingClass = new CodeClass(currentNamespace) { Name = className, ClassKind = CodeClassKind.Model };
+                if(inheritsFrom == null && schema.AllOf.Count > 1) { //the last is always the current class, we want the one before the last as parent
```

Comment:
There are multiple ways of expressing inheritance in JSON Schema.  We will need to add support for other ways later.

Reply:
Can you log an issue with the details please?